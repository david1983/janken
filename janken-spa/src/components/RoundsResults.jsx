import React from 'react';
import PropTypes from 'prop-types';
import { observer, inject } from 'mobx-react';
import {
  Table, TableHead, TableBody, TableRow, TableCell,
} from '@material-ui/core';

import GameClass from '../state/Game';

/**
 * Component that renders a table with the results
 * for each round of the game
 * @param {Object} param0 the component props
 */
const RoundsResults = ({ Game }) => (

  <Table>
    <TableHead>
      <TableRow>
        {Game.players.map((i, k) => (
          <TableCell
            key={`header-${k}`}
          >
            {`Player ${k + 1} - ${i.name}`}
          </TableCell>
        ))}
        <TableCell>Winner</TableCell>
      </TableRow>
    </TableHead>
    <TableBody>
      {Game.rounds.map((round, idx) => (
        <TableRow key={`round-${idx}`}>
          {Object.keys(round.players)
            .map(playerKey => (
              <TableCell
                key={playerKey}
              >
                {round.players[playerKey].move}
              </TableCell>
            ))}
          <TableCell>{round.winner}</TableCell>
        </TableRow>
      ))}
    </TableBody>
  </Table>
);

RoundsResults.defaultProps = {
  Game: new GameClass(),
};

RoundsResults.propTypes = {
  Game: PropTypes.object,
};


export default inject('Game')(observer(RoundsResults));
