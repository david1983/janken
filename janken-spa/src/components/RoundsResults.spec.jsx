import React from 'react';
import { render } from 'enzyme';
import RoundsResults from './RoundsResults';
import GameClass from '../state/Game';

const Game = new GameClass();
describe('the RoundsResults component', () => {
  it('should match the snapshot of the shallow rendered component', () => {
    const component = <RoundsResults Game={Game} />;
    const rendered = render(component);
    expect(rendered).toMatchSnapshot();
  });
});
