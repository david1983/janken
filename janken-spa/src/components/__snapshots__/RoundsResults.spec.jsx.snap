// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the RoundsResults component should match the snapshot of the shallow rendered component 1`] = `
<table
  class="MuiTable-root"
>
  <thead
    class="MuiTableHead-root"
  >
    <tr
      class="MuiTableRow-root MuiTableRow-head"
    >
      <th
        class="MuiTableCell-root MuiTableCell-head"
        scope="col"
      >
        Player 1 - 
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head"
        scope="col"
      >
        Player 2 - 
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head"
        scope="col"
      >
        Winner
      </th>
    </tr>
  </thead>
  <tbody
    class="MuiTableBody-root"
  />
</table>
`;
